packages/web-components/fast-components/src/styles/patterns/button.ts

```diff
@@ -38,457 +45,418 @@ import {
-        color: inherit;
-        border-radius: inherit;
-        fill: inherit;
-        cursor: inherit;
-        font-family: inherit;
-        font-size: inherit;
-        line-height: inherit;
-    }
-
-    :host(:hover) {
-        background-color: ${neutralFillHover};
-    }
-
-    :host(:active) {
-        background-color: ${neutralFillActive};
-    }
-
-    .control:${focusVisible} {
-        border-color: ${focusStrokeOuter};
-        box-shadow: 0 0 0 calc((${focusStrokeWidth} - ${strokeWidth}) * 1px) ${focusStrokeOuter} inset;
-    }
-
-    .control::-moz-focus-inner {
-        border: 0;
-    }
-
-    .start,
-    .end {
-        display: flex;
-    }
-
-    .control.icon-only {
-        padding: 0;
-        line-height: 0;
-    }
-
-    ::slotted(svg) {
-        ${
-            /* Glyph size and margin-left is temporary -
-            replace when adaptive typography is figured out */ ""
-        } width: 16px;
-        height: 16px;
-        pointer-events: none;
-    }
-
-    .start {
-        margin-inline-end: 11px;
-    }
-
-    .end {
-        margin-inline-start: 11px;
-    }
-`.withBehaviors(
-    forcedColorsStylesheetBehavior(
-        css`
-            :host .control {
-              background-color: ${SystemColors.ButtonFace};
-              border-color: ${SystemColors.ButtonText};
-              color: ${SystemColors.ButtonText};
-              fill: currentColor;
-            }
-    
-            :host(:hover) .control {
-              forced-color-adjust: none;
-              background-color: ${SystemColors.Highlight};
-              color: ${SystemColors.HighlightText};
-            }
-
-            .control:${focusVisible} {
-              forced-color-adjust: none;
-              background-color: ${SystemColors.Highlight};
-              border-color: ${SystemColors.ButtonText};
-              box-shadow: 0 0 0 calc((${focusStrokeWidth} - ${strokeWidth}) * 1px) ${SystemColors.ButtonText} inset;
-              color: ${SystemColors.HighlightText};
-            }
-
-            .control:hover,
-            :host([appearance="outline"]) .control:hover {
-              border-color: ${SystemColors.ButtonText};
-            }
-
-            :host([href]) .control {
-                border-color: ${SystemColors.LinkText};
-                color: ${SystemColors.LinkText};
-            }
-
-            :host([href]) .control:hover,
-            :host([href]) .control:${focusVisible}{
-              forced-color-adjust: none;
-              background: ${SystemColors.ButtonFace};
-              border-color: ${SystemColors.LinkText};
-              box-shadow: 0 0 0 1px ${SystemColors.LinkText} inset;
-              color: ${SystemColors.LinkText};
-              fill: currentColor;
-            }
-        `
-    )
-);
+export const BaseButtonStyles = (
+    context: ElementDefinitionContext,
+    definition: FoundationElementDefinition,
+    interactivitySelector: string = ""
+) =>
```

Comment:
You are calling out the opposite of the interactivitySelector explicitly in buton.styles.ts and anchor.styles.ts above. Would it make sense to create that here and move the styles into BaseButtonStyles?

```
const inactiveSelector = interactivitySelector.indexOf(":not") >= 0 ? 
    interactivitySelector.replace(/(.*):not\(([^)]*)\)(.*)/, '$1$2$3') : 
    `:not(${interactivitySelector})`; 
```

Reply:
updated. thanks